_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,324 @@
+			<td>8:00</td>
+			<td>13:00</td>
+			<td>18:00</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 5
+
+Each column header is assigned to a cell. Usage of `headers` attribute changes the relationship between column headers and cells.
+
+```html
+<table>
+	<tr>
+		<th id="col1">Column 1</th>
+		<th id="col2">Column 2</th>
+	</tr>
+	<tr>
+		<td></td>
+	</tr>
+	<tr>
+		<td headers="col2"></td>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The column header ("Column 2"), does not have an assigned cell within the same `table` element.
+
+```html
+<table>
+	<thead>
+		<tr>
+			<th>Column 1</th>
+			<th>Column 2</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td>15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Failed Example 2
+
+The column header ('Column2') does not have an assigned cell within the same `table` element. In this case the usage of `headers` attribute removes cell association to the column.
+
+```html
+<table>
+	<tr>
+		<th id="col1"></th>
+		<th id="Column2"></th>
+	</tr>
+	<tr>
+		<td></td>
+		<td headers="col1"></td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
+
+The column header ('Column B') does not have an assigned cell within the same `table` element.
+
+```html
+<div role="grid">
+	<div role="row">
+		<div role="columnheader">Col A</div>
+		<div role="columnheader">Col B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">1A</div>
+		<div role="gridcell">1B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">2A</div>
+	</div>
+</div>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+The only element with a header [role][semantic role] is part of a table which is not [included in the accessibility tree][].
+
+```html
+<table role="presentation">
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>12:00</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 2
+
+The only element with header [role][semantic role] is part of a table which is not [visible][].
+
+```html
+<html>
+	<style>
+		.notInPage {
+			position: absolute;
+			left: -9999px;
+			top: -9999px;
+		}
+	</style>
+	<body>
+		<table class="notInPage">
+			<tr>
+				<th>Time</th>
+			</tr>
+			<tr>
+				<td>24:00</td>
+			</tr>
+		</table>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 3
+
+There is no element with a [semantic role][] of header within the table.
+
+```html
+<table>
+	<tr>
+		<td>12:00</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 4
+
+There are no [descendants][descendant] within the table.
+
+```html
+<table></table>
+```
+
+#### Inapplicable Example 5
+
+The only element with a [role][semantic role] of header is not [visible][].
+
+```html
+<table>
+	<tr>
+		<th style="display: none;"></th>
+	</tr>
+	<tr>
+		<td></td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 6
+
+There are no elements with a [role][semantic role] of header within the table.
+
+```html
+<table>
+	<tr>
+		<th role="cell"></th>
+	</tr>
+	<tr>
+		<td></td>
+	</tr>
+</table>
+```
+
```

Comment:
Need an  Inapplicable example where the only header are not included in the accessibility tree.